### PR TITLE
Fixes #3 -- generate user task links

### LINKED ISSUE #3: Set up endpoint to generate magic links
Reference: `django.contrib.auth.tokens` and the various forms/views related to password
resets in `django.contrib.auth`.

The magic, secure link must be generated by the ZAC lite itself, since it usually uses
the `settings.SECRET_KEY` which must stay private to ZAC lite.

Request:

```http
POST http://localhost:8000/api/v1/user-link HTTP/1.1
Host: localhost:8000
Content-Type: application/json

{
    "taskId": "753e682d-b9af-4efa-811f-a2c8b0b51967"
}
```

Response:

```json
{
    "url": "http://localhost:8000/ui/perform-task/<tidb64>/<token>"
}
```

Where `tidb64` is the urlsafe base64 encoded task ID, and the token the
cryptographically secure token.

The token should use as hash input:

* `Task.assignee`
* `Task.due`
* `Task.delegationState`
* `Task.owner`
* `Task.suspended`
* `Task.formKey`

If any of those task properties change after the token/URL is generated, the token will
be invalidated.

This endpoint will be called by BPTL work units in the Camunda process, so it should
have `TokenAuthentication` auth backend.

**Acceptance criteria**

- [x] Endpoint implemented
- [x] Endpoint tested
- [x] Endpoint is documented
- [x] Only `TokenAuthentication` supported
- [x] Token varies with specified task parameters
- [x] URL is fully qualified, absolute URL

### PR DESCRIPTION
**Pull request by sergei-maertens**
This endpoint is meant to be called from Camunda or other tools to generate a frontend URL.

The end-user can use this URL to access the task, without needing to log in. The URL is secured using a salted HMAC token, and the token invalidates in the required conditions.

### CODE REVIEW

File: backend/src/zac_lite/user_tasks/serializers.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        ui_path = FRONTEND_URL.format(tidb64=tidb64, token=token)
+        return self.request.build_absolute_uri(ui_path)
+
+
+class UserLinkSerializer(serializers.Serializer):
+    task_id = serializers.UUIDField(
+        required=True,
+        label=_("Camunda Task ID"),
+        help_text=_(
+            "The ID of the user task in Camunda, which is normally in the UUID 4 format."
```
Comment (damm89) [inline backend/src/zac_lite/user_tasks/serializers.py:42]: _Required_ and _normally_ and _UUIDField_ kind of set off an alarm bell. Will the abnormal cases also work?

File: backend/src/zac_lite/conf/base.py [hunk 2]
```diff
@@ -341,6 +341,11 @@
 ENVIRONMENT = config("ENVIRONMENT", "")
 SHOW_ALERT = True
 
+#
+# EXECUTE TASK TOKEN INVALIDATION
+#
+EXECUTE_TASK_TOKEN_TIMEOUT_DAYS = config("EXECUTE_TASK_TOKEN_TIMEOUT_DAYS", default=7)
```
Comment (damm89) [inline backend/src/zac_lite/conf/base.py:347]: Why 7 days? Is that going to be our cleaning frequency? Would it be desirable to have different timeout durations for different tasks?

File: backend/src/zac_lite/user_tasks/serializers.py (hunk 1 above)
Comment (sergei-maertens) [reply backend/src/zac_lite/user_tasks/serializers.py:42]: So far I haven't seen cases where it isn't a UUID. The API documentation of Camunda says it's a string, we observe that the format UUID applies. With their latest version they also specify an OpenAPI documentation, but you need to extract it from their jars it seems so I haven't been able to verify that they also specify it as UUID. We can easily relax it to just CharField if it needs to be though.

File: backend/src/zac_lite/conf/base.py (hunk 2 above)
Comment (sergei-maertens) [reply backend/src/zac_lite/conf/base.py:347]: Yeah, 7 days seems now a reasonable action time but it's just a default now that we can easily customize through the environment with input from FB.

There is a TODO in the token generator to indeed be able to customize the timeout duration per task, but then we would need to start storing task data + timeouts in the database which might not be desirable. For an initial pass, I think a week should suffice.

**Review by damm89 — COMMENTED** [review]
Nice. :)

**Pull request merged**